website/docs/optimizations/java/optimized-jvm.md

```diff
@@ -0,0 +1,154 @@
+```
+
+As discussed in the first chapter the Dockerfile leverages a multi-stage build. In the first stage (starting line 1), the application and a custom runtime is built. The custom runtime is based on the list of dependencies that have been collected with `jdeps` (see line 11 - 14).
+
+In the first step, the complete classpath of the application is analyzed and all module dependencies are written in a file called `jre-deps.info`. You may have noticed the explicit addition of `jdk.crypto.ec`. This module provides the implementation of the SunEC security provider which is necessary for TLS support. Unfortunately, it is not obvious from the analysis of the classpath that this additional module is necessary. For this reason, it is added at this point.
+
+The `jre-deps.info` is used as input for `jlink` (line 20 - 23) in order to build a custom runtime. The goal of this is to reduce the size as much as possible, that's the reason the runtime is compressed and no header files and no man-pages are included.
+
+In the second stage of our build (line 28 - 29), we copy the JAR file of the application as well the custom runtime to the target image.
+
+## 3. Re-deploying the application
+
+After the new image has been pushed to ECR you can re-trigger the deployment of the application:
+
+```bash
+kubectl rollout restart deploy unicorn-store-spring -n unicorn-store-spring
+kubectl rollout status deployment unicorn-store-spring -n unicorn-store-spring
+```
+
+## 4. Testing the application
+
+Run the following API call to verify that the new version of the application has been deployed successfully:
+
+1. Export the Service URL for later use:
+
+```bash showLineNumbers
+export SVC_URL=http://$(kubectl get svc unicorn-store-spring -n unicorn-store-spring -o json | jq --raw-output '.status.loadBalancer.ingress[0].hostname')
+curl --location --request GET $SVC_URL'/' --header 'Content-Type: application/json'; echo
+```
+
+![optimized-jvm-result](./images/optimized-jvm-result.png)
+
+## 5. Retrieving the results
+
+Verify the new application image size in the [Amazon ECR](https://console.aws.amazon.com/ecr/home#/) console:
+
+![optimized-jvm-ecr](./images/optimized-jvm-ecr.png)
+
+Retrieve the logs for the application as outlined in the previous section. Below you can find an example starting time after the optimization of the Amazon EKS deployment:
+
+```bash showLineNumbers
+kubectl logs $(kubectl get pods -n unicorn-store-spring -o json | jq --raw-output '.items[0].metadata.name') -n unicorn-store-spring
+kubectl logs $(kubectl get pods -n unicorn-store-spring -o json | jq --raw-output '.items[0].metadata.name') -n unicorn-store-spring | grep "Started StoreApplication"
+```
+
+![optimized-jvm-eks](./images/optimized-jvm-eks.png)
+
+## Conclusion
+
+As you can see, we were able to reduce the size of the container image from 380 MB to 257 MB. The application startup time improved slightly.
```

Comment:
As you can see, we managed to decrease the size of the container image from 380 MB to 257 MB. The application startup time also improved by a small margin.